regcomp.c

```diff
@@ -12550,6 +12550,37 @@ S_regbranch(pTHX_ RExC_state_t *pRExC_state, I32 *flagp, I32 first, U32 depth)
     return ret;
 }
 
+U32
+S_get_quantifier_value(pTHX_ RExC_state_t *pRExC_state,
```

Comment:
I don't understand what this function is doing: why is a null byte "invalid" while nothing else is? Why do we touch `RExC_parse` in the failure cases? Should it be checking that `end` is unchanged by grok_atoUV?

This feels like it is very tightly coupled to what regcurly and regpiece do, in subtle ways.

Reply:
I should have commented what this function does, and am abashed that I didn't.
That null should be '0' a zero character.  I will add a test to the suite for this case.
Perhaps there should be at least an assert that 'end' doesn't get changed, but since it is tightly coupled with regpiece(), there shouldn't be a possibility of this.

It is a common paradigm throughout regcomp.c to position RExC_parse prior to outputting a diagnostic so that the <--HERE in that diagnostic comes out in the proper place.  It's not clear to me that every instance of this needs to have a comment about that.

Reply:
And I think the position of the HERE in these diagnostics is wrong, but wanted to start with a state where no external behavior change is visible.  I think it should be ...}<--HERE. And I intend to make a later commit change to do that, unless there is a good reason not to.